src/analysis/loadGraph.test.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+
+import tmp from "tmp";
+import path from "path";
+
+import {Graph, NodeAddress, EdgeAddress} from "../core/graph";
+import type {IAnalysisAdapter} from "../analysis/analysisAdapter";
+import * as RepoIdRegistry from "../core/repoIdRegistry";
+import {makeRepoId, type RepoId} from "../core/repoId";
+import {loadGraph} from "./loadGraph";
+
+class MockAnalysisAdapter implements IAnalysisAdapter {
+  _resolutionGraph: ?Graph;
+  _name: string;
+
+  /**
+   * Takes a name for the plugin, and a graph that
+   * is provided as a result of a successful load.
+   * If no graph is provided, then load will fail.
+   */
+  constructor(name: string, resolutionGraph: ?Graph) {
+    this._name = name;
+    this._resolutionGraph = resolutionGraph;
+  }
+
+  declaration() {
+    return {
+      name: this._name,
+      nodePrefix: NodeAddress.empty,
+      edgePrefix: EdgeAddress.empty,
+      nodeTypes: [],
+      edgeTypes: [],
+    };
+  }
+
+  async load(
+    _unused_sourcecredDirectory: string,
+    _unused_repoId: RepoId
+  ): Promise<Graph> {
+    if (this._resolutionGraph != null) {
+      return this._resolutionGraph;
+    } else {
+      throw new Error("MockAnalysisAdapterRejects");
+    }
+  }
+}
+
+describe("analysis/loadGraph", () => {
+  function setUpRegistryWithId(repoId: RepoId) {
+    const dirname = tmp.dirSync().name;
+    process.env.SOURCECRED_DIRECTORY = dirname;
+    const registry = RepoIdRegistry.addEntry(RepoIdRegistry.emptyRegistry(), {
+      repoId,
+    });
+    RepoIdRegistry.writeRegistry(registry, dirname);
+    return dirname;
+  }
+  describe("loadGraph", () => {
```

Comment:
Good catch! It's a very realistic scenario. Because I know the contract of RepoIdRegistry, I know that these cases where the registry doesn't exist will return an empty registry, which matches the case you described. However, it will be better to test it explicitly, rather than implicitly depending on RepoIdRegistry. Updated to add an additional test case.